app/models/manageiq/providers/infra_manager/provision_workflow.rb

```diff
@@ -1,11 +1,12 @@
 class ManageIQ::Providers::InfraManager::ProvisionWorkflow < ::MiqProvisionVirtWorkflow
   def set_or_default_hardware_field_values(vm)
     update_values = {
-      :vm_memory      => vm.hardware.memory_mb.to_s,
-      :cpu_limit      => vm.cpu_limit,
-      :memory_limit   => vm.memory_limit,
-      :cpu_reserve    => vm.cpu_reserve,
-      :memory_reserve => vm.memory_reserve
+      :vm_memory              => vm.hardware.memory_mb.to_s,
+      :cpu_limit              => vm.cpu_limit,
+      :memory_limit           => vm.memory_limit,
+      :cpu_reserve            => vm.cpu_reserve,
+      :memory_reserve         => vm.memory_reserve,
+      :allocated_disk_storage => vm.hardware.disks.find_by(:device_type => "disk").size / (1024**3)
```

Comment:
I know in the VMware PR we're discussing finding the "first" disk as an initial solution, that is relatively simple for a specific provider but it is going to be very tricky here since this code is common to all Infra providers.

In `MiqProvisionVirtWorkflow#set_on_vm_id_changed` we build a map of the NICs and LANs: https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_provision_virt_workflow.rb#L137-L138

I wonder if we do the same with the template's disks and then we could more easily set a different size for each disk.

Reply:
I think we should change this, but speaking of this code specifically:
1. Prefer `1.gigabyte` over `1024**3` for readability
2. It would be better tot pass around `bytes` if we have it rather than `GB`, we can always convert to GB but not the other way around without losing precision (also if we're going to use `GB` should be a float e.g. if a user wants `1.5GB` disk)

Reply:
I changed the field to string and now the user can input a float-like value.
regarding `bytes`, the `.size` attribute of a disk is indeed in `bytes`, but is converted to a human readable GB to provide the default value, and is received from the user in GB scale as well. 
afterwards, `VirtualDisk` reconfiguration expects it in KB for `dev.capacityInKB` (in `ManageIQ::Providers::Vmware::InfraManager::Provision::Configuration::Disk#set_disk_allocation`).
so I'm not sure where in that flow will it be effective to convert it to `bytes`

Reply:
> I know in the VMware PR we're discussing finding the "first" disk as an initial solution, that is relatively simple for a specific provider but it is going to be very tricky here since this code is common to all Infra providers.
> 
> In `MiqProvisionVirtWorkflow#set_on_vm_id_changed` we build a map of the NICs and LANs: https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_provision_virt_workflow.rb#L137-L138
> 
> I wonder if we do the same with the template's disks and then we could more easily set a different size for each disk.

in line with the short-fix you suggested in the vmware repo PR, the code now adds the default `:allocated_disk_storage` only when there's exactly 1 hard-disk.